### DISCUSSION TITLE
Extending existing modules (ex: jetty-core)

### DISCUSSION THREAD

**Discussion by ricardas-buc** [eclipse-edc/Connector#3964] [Q&A] [answered]
For our use cases we require customized jetty https connector (we need use custom SSLContext), but I am not seeing a good way of extending JettyService class without duplicating a lot of code and doing it in a "hackish" way.

What would be the go-to way to customize it?

**Comment by jimmarino** [comment]
Just replace the module with your implementation and exclude it from your distribution build.

**Reply by ricardas-buc** [reply]
Thanks for a quick reply!
That sounds fine as a stop-gap solution, but this would just add unnecessary complexity when we want to upgrade edc.

jetty-core on dependency graph is also deep inside, it is pulled in by jersey-core and it is used by every api module. If there is no other way I will try excluding it, but it sounds that it will be rather tedious.

**Reply by jimmarino** [reply, marked as answer]
Globally excluding the dependency is straightforward in build systems such a Gradle. 

The Jetty service can be replaced, but it is not designed to be extensible. Introducing extensibility to this service will add complexity to core EDC for use cases that are not common. If you need this type of control over a low-level service, you will need to replace it.

**Reply by ndr-brt** [reply]
> Thanks for a quick reply! That sounds fine as a stop-gap solution, but this would just add unnecessary complexity when we want to upgrade edc.
> 
> jetty-core on dependency graph is also deep inside, it is pulled in by jersey-core and it is used by every api module. If there is no other way I will try excluding it, but it sounds that it will be rather tedious.

its dependency on `jersey-core` I think it's a refuse and it can be removed, there are also others of it that can be cleaned up, but surely management-api modules do not define it directly that (if so it is for tests).
Eventually you can always [exclude dependencies with gradle](https://docs.gradle.org/current/userguide/dependency_downgrade_and_exclude.html#sec:excluding-transitive-deps)
